File: Models/Responses/Users/ResponseUserRole.cs
namespace Models.Responses.Users;

/// <summary>
/// 사용자 역할 클래스 
/// </summary>
public class ResponseUserRole
{
    /// <summary>
    /// 역할명 
    /// </summary>
    public required string Name { get; set; }

    /// <summary>
    /// 역할 상세 Claim
    /// </summary>
    public List<ResponseUserRoleClaim>? Claims { get; set; }
}

File: BudgetBringerV2/src/models/responses/users/response-user-role.ts
import type { ResponseUserRoleClaim } from '@/models/responses/users/response-user-role-claim'

/**
 * 사용자 역할 클래스
 */
export interface ResponseUserRole {
  /**
   * 역할 명
   */
  name : string;
  /**
   * 상세 권한
   */
  claims : Array<ResponseUserRoleClaim>;
}
